src/backends/wasm_js.rs

```diff
@@ -7,34 +7,54 @@ pub use crate::util::{inner_u32, inner_u64};
 #[cfg(not(all(target_arch = "wasm32", any(target_os = "unknown", target_os = "none"))))]
 compile_error!("`wasm_js` backend can be enabled only for OS-less WASM targets!");
 
-use js_sys::{global, Uint8Array};
-use wasm_bindgen::{prelude::wasm_bindgen, JsCast, JsValue};
+#[cfg(target_feature = "atomics")]
+use js_sys::Uint8Array;
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue};
 
-// Size of our temporary Uint8Array buffer used with WebCrypto methods
-// Maximum is 65536 bytes see https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues
-const CRYPTO_BUFFER_SIZE: u16 = 256;
+// Maximum buffer size allowed in `Crypto.getRandomValuesSize` is 65536 bytes.
+// See https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues
+const MAX_BUFFER_SIZE: u32 = 65536;
```

Comment:
I think it's better to use `usize` here since it will allow us to remove several `as` casts.

Reply:
Unfortunately it won't save us from casting. We need it in `u32` for all `Uint8Array` methods. So we would just be trading one set of `as` casts with another.

I tried out adding `const MAX_BUFFER_USIZE: usize = MAX_BUFFER_SIZE as usize;` which is checked during compile time. But I found it a bit strange.

Let me know if you like that idea.

Reply:
You can write the following:
```rust
let buf_len = usize::min(dest.len(), MAX_BUFFER_SIZE);
let buf_len_u32 = buf_len.try_into().expect("...");
let buf = Uint8Array::new_with_length(buf_len_u32);
for chunk in dest.chunks_mut(buf_len) { ... }
```

Number of `TryInto` casts stays the same and you can remove two `as` casts.

Reply:
Nice! We have 0 `as` casts now!